src/crossval.jl

```diff
@@ -11,9 +11,9 @@ struct Kfold <: CrossValGenerator
     k::Int
     coeff::Float64
 
-    function Kfold(n::Int, k::Int)
+    function Kfold(n::Int, k::Int, rng::AbstractRNG = MersenneTwister())
```

Comment:
The general convention seems to be to put the RNG argument first for methods that accept one. I guess it's a bit less clear whether it applies in this case given that it's for a constructor for a type that doesn't hold onto the RNG but I think it'd still probably be good to follow the general pattern. It is admittedly more annoying though because there isn't convenient syntax for it. But to that end, you could do
```julia
function Kfold(rng::AbstractRNG, n::Int, k::Int)
    # existing contents
end
Kfold(n::Int, k::Int) = Kfold(MersenneTwister(), n, k)
```
and likewise for `StratifiedKfold`.

`RandomSub` is interesting though because it does hold onto the RNG. I guess it should stay in the corresponding position then, which means that it's different from `(Stratified)Kfold`. Maybe that's okay though? Curious to hear your thoughts on that.

I also wonder whether there could be any benefits to having `(Stratified)Kfold` also hold onto the RNG.

Reply:
I think this is a very important detail that never occurred to me. If this is the general convention, I think it would be good to adhere to it. 

One question: what do you mean with "holding onto" the RNG? I suppose you mean storing it in a field, right?

My (non-expert) opinion would be then to indeed put the RNG as the first argument in `Kfold`, `StratifiedKfold` and in `RandomSub`. 

I am very happy to change the code accordingly. Should I push additional changes to this pull request?

Reply:
> what do you mean with "holding onto" the RNG? I suppose you mean storing it in a field, right?

Ah sorry about that, my choice of words was not particularly clear. 😅 Your interpretation is correct, I meant storing the RNG object as one of the fields.

> My (non-expert) opinion would be then to indeed put the RNG as the first argument in `Kfold`, `StratifiedKfold` and in `RandomSub`.

Just to clarify, you would have them all take the RNG as the first argument, even though for `RandomSub` that doesn't correspond to position in the fields?

Another thing that occurred to me is that typically the default value for RNG arguments is `Random.GLOBAL_RNG` but currently the PR uses `MersenneTwister()`. I think it would probably be good to use the global RNG as the default, which will also match the behavior of these constructors as they exist on `master` currently. Or was the choice of `MersenneTwister` deliberate?

Reply:
First of all many thanks for your time and valuable input. This is my first ever contribution to a Julia package. I thought I should start with tackling something relatively easy and did not anticipate all these important details to crop up. I hope in the future I can make more careful and efficient contributions ;)

> Just to clarify, you would have them all take the RNG as the first argument, even though for `RandomSub` that doesn't correspond to position in the fields?

Indeed, I would have all of them taking RNG as the first argument including `RandomSub`. For `RandomSub`, I would move the corresponding newly introduced in the commit field `rng::AbstractRNG` to the first position. I hope this is clearer now.

> Another thing that occurred to me is that typically the default value for RNG arguments is `Random.GLOBAL_RNG` but currently the PR uses `MersenneTwister()`. I think it would probably be good to use the global RNG as the default, which will also match the behavior of these constructors as they exist on `master` currently. Or was the choice of `MersenneTwister` deliberate?

There is absolutely no deliberation behind the choice of `MersenneTwister`. Again, I think we should use `Random.GLOBAL_RNG` as the default option as you suggest.

Reply:
> First of all, many thanks for your time and valuable input.

I'm more than happy to help, I just hope my comments have been more useful than annoying. 😅 This is looking great so far and I appreciate the contribution!

> I hope this is clearer now.

👍 Yes indeed, thanks!

That all sounds good to me. Would you be willing to make those changes in this PR then?

Reply:
Yes, I am very happy to do so. I will get on with it early next week.